cibuildwheel/pyodide.py

```diff
@@ -165,13 +221,28 @@ def setup_python(
         env=env,
     )
 
-    log.step("Installing emscripten...")
+    log.step(f"Installing Emscripten version: {python_configuration.emscripten_version} ...")
     emcc_path = install_emscripten(tmp, python_configuration.emscripten_version)
 
     env["PATH"] = os.pathsep.join([str(emcc_path.parent), env["PATH"]])
 
-    log.step("Installing Pyodide xbuildenv...")
-    env["PYODIDE_ROOT"] = install_xbuildenv(env, python_configuration.pyodide_version)
+    # Allow overriding the xbuildenv version with an environment variable. This allows
+    # testing new Pyodide xbuildenv versions before they are officially released, or for
+    # using a different Pyodide version other than the one that is listed to be compatible
+    # with the pyodide-build version in the build-platforms.toml file.
+    cibw_pyodide_version = os.environ.get(
+        "CIBW_PYODIDE_VERSION", python_configuration.pyodide_version
+    )
```

Comment:
The other options can be passed either as a cli flag e.g., `--pyodide-version` or as the corresponding environment variable in this case `CIBW_PYODIDE_VERSION`. They can also come from the cibuildwheel section in `pyproject.toml`. We should handle this in the same way as all the other options. Then we can just use `python_configuration.pyodide_version` and it should already have the right value.

Reply:
Most options are not available as flags (and I wouldn't add this one), only stuff you often need on the command line, like `--platform`. Just env var and config.